### ISSUE TITLE
Console.In.Peek() always returning EOF after the first read line on Windows

### ISSUE THREAD

**Issue by eiriktsarpalis** [dotnet/runtime#40735]
I came across the following behaviour when debugging a System.Console test. Running
```csharp
static void Main()
{
    while (true)
    {
        Console.Write("Enter some text: ");
        int c = Console.In.Peek();
        string? line = Console.In.ReadLine();
        Console.WriteLine($"Peek: {c} ReadLine: {line}");
    }
}
```
Produces the following behaviour
```
Enter some text: test
Peek: 116 ReadLine: test
Enter some text: test
Peek: -1 ReadLine: test
Enter some text: test
Peek: -1 ReadLine: test
```
Essentially it is impossible to use `Console.In.Peek()` after the first line has been consumed. Note that this behaviour manifests itself regardless of whether the buffer contains any characters. The following variant will produce the exact same observable behaviour:
```csharp
static void Main()
{
    while(true)
    {
        Console.Write("Enter some text: ");
        int c = MyPeek();
        string? line = Console.In.ReadLine();
        Console.WriteLine($"Peek: {c} ReadLine: {line}");
    }

    static int MyPeek()
    {
        while (!Console.KeyAvailable)
        {
            Thread.Sleep(10);
        }
        return Console.In.Peek();
    }
}
```
Note that unix does not have the same behaviour.

**Comment by eiriktsarpalis** [comment]
Related to #39192.

**Comment by adamsitnik** [comment]
I assume that this is not a regression and moving it to future

**Comment by eiriktsarpalis** [comment]
> I assume that this is not a regression and moving it to future

That's correct, this seems to have been the behaviour for quite a while. The main goal here is achieving consistency with unix behaviour.

**Comment by caraioniurie47** [comment]
This no longer reproduces: it was fixed by #89609 (merged 2023-07-29, fixing #68983), which shipped in .NET 8.

Before that change, `StreamReader.Peek()` returned -1 without reading whenever its buffer was empty and `_isBlocked` was set (`if (_isBlocked || ReadBuffer() == 0)`), and `_isBlocked` was set whenever a read returned fewer bytes than the buffer (`_isBlocked = (_byteLen < _byteBuffer.Length);`). `Console.In` is created with a 4096-byte buffer and a console read returns a single line, so after the first line every `Peek()` returned -1. #89609 removed the `_isBlocked ||` condition from `Peek()`.

Both programs from the issue description, run for three lines each, on Windows 11 build 26200:

| runtime | inbox conhost | OpenConsole 1.0.200517002 (2020) |
|---|---|---|
| .NET Core 3.1.4 | `Peek: 116`, then `-1`, `-1` | same |
| .NET 10.0.12 | `Peek: 116` on all three lines | same |

`ReadLine` returned `test` every time on both runtimes. The console host makes no difference, so this was a runtime bug rather than a host one.

The `[ActiveIssue("https://github.com/dotnet/runtime/issues/40735", TestPlatforms.Windows)]` on `ConsoleManualTests.InPeek` could be removed as well. Its Peek/Peek/Read sequence, run after a `ReadLine()`, gets -1 for the first character on .NET Core 3.1.4 and the expected character throughout on .NET 10.0.12.

This issue could be closed by a PR removing the `[ActiveIssue]`. I'll send that PR; could this be assigned to me?

> [!NOTE]
> AI-generated, written at my direction and reviewed by me before posting. Source read at commit `60c49c6156e` and its parent. The programs were built as `netcoreapp3.1` and `net10.0` and run on the shipped .NET Core 3.1.4 and .NET 10.0.12 runtimes, with the keystrokes written into their console input from a separate process (`AttachConsole` + `WriteConsoleInput`) and the screen buffer read back from there. The `InPeek` sequence was run the same way, under inbox conhost only; the manual test itself was not run.
